File: src/market-tracker-webapi/Application/Domain/Schemas/Market/Inventory/Product/ProductReview.cs
using market_tracker_webapi.Application.Domain.Schemas.Account.Users;

namespace market_tracker_webapi.Application.Domain.Schemas.Market.Inventory.Product;

public record ProductReview(
    ReviewId Id,
    ClientItem Client,
    string ProductId,
    int Rating,
    string? Comment,
    DateTime CreatedAt
)
{
    public ProductReview(
        int Id,
        ClientItem Client,
        string ProductId,
        int Rating,
        string? Comment,
        DateTime CreatedAt
    ) : this(
        new ReviewId(Id),
        Client,
        ProductId,
        Rating,
        Comment,
        CreatedAt
    )
    {
    }
};

public record ReviewId(
    int Value
);

File: src/market-tracker-android/app/src/main/java/pt/isel/markettracker/domain/model/market/inventory/product/ProductReview.kt
package pt.isel.markettracker.domain.model.market.inventory.product

import pt.isel.markettracker.domain.model.account.ClientItem
import java.time.LocalDateTime

data class ProductReview(
    val id: Int,
    val productId: String,
    val rating: Int,
    val comment: String?,
    val createdAt: LocalDateTime,
    val client: ClientItem
)
